tracex_project/extraction/tests.py

```diff
@@ -0,0 +1,169 @@
+
+from extraction.logic.orchestrator import Orchestrator, ExtractionConfiguration
+from extraction.logic.modules.module_activity_labeler import ActivityLabeler
+from extraction.logic.modules.module_time_extractor import TimeExtractor
+from extraction.logic.modules.module_event_type_classifier import EventTypeClassifier
+from extraction.logic.modules.module_location_extractor import LocationExtractor
+
+
+class OrchestratorTests(TestCase):
+    """Test cases for the Orchestrator class utilizing the ExtractionConfiguration."""
+
+    def test_single_instance_creation(self):
+        """Tests if two initialized orchestrators are the same instance."""
+        Orchestrator.reset_instance()
+        orchestrator1 = Orchestrator()
+        orchestrator2 = Orchestrator()
+        self.assertIs(orchestrator1, orchestrator2)
+
+    def test_consistent_object_state(self):
+        """Tests if the state of the orchestrator instance is the same for two instances."""
+        Orchestrator.reset_instance()
+        orchestrator1 = Orchestrator()
+        orchestrator2 = Orchestrator()
+
+        orchestrator1.data = 'test_data'
+
+        self.assertEqual(orchestrator1.data, orchestrator2.data)
+
+    def test_get_instance_method(self):
+        """Tests if the get_instance method returns the same instance and if a new instance is the same instance."""
+        Orchestrator.reset_instance()
+        Orchestrator()
+        orchestrator1 = Orchestrator.get_instance()
+        orchestrator2 = Orchestrator.get_instance()
+
+        self.assertIs(orchestrator1, orchestrator2)
+
+        orchestrator3 = Orchestrator()
+        self.assertIs(orchestrator1, orchestrator3)
+
+    def test_reset_instance(self):
+        """Tests if reset_instance resets the instance for all objects."""
+        Orchestrator.reset_instance()
+        orchestrator1 = Orchestrator()
+        Orchestrator.reset_instance()
+        orchestrator2 = Orchestrator
+        self.assertIsNot(orchestrator1, orchestrator2)
+
+    def test_set_configuration(self):
+        """Tests if the set_configuration method correctly updates the Orchestrators instance's configuration."""
+        Orchestrator.reset_instance()
+        config = ExtractionConfiguration()
+        orchestrator = Orchestrator(config)
+        new_config = ExtractionConfiguration
+        orchestrator.set_configuration(new_config)
+        self.assertIs(orchestrator.configuration, new_config)
+
+    def test_singleton_with_configuration(self):
+        """Tests that the Orchestrator's configuration remains unchanged with subsequent instantiations."""
+        Orchestrator.reset_instance()
+        config1 = ExtractionConfiguration()
+        orchestrator1 = Orchestrator(config1)
+        config2 = ExtractionConfiguration()
+        orchestrator2 = Orchestrator(config2)
+        self.assertIs(orchestrator1.configuration, orchestrator2.configuration)
+
+    # Does not work yet, need to adjust initialization of modules

```

Comment:
What exactly does not work? Maybe we should put this test in comment or otherwise disable it so it is not executed when running all tests until it works as intended.

Reply:
The purpose of this test case is to test the initialize_modules method from the orchestrator. Therefore I want to initialize specific modules and test the returned list from the initialize_modules method contains them. I have changed the test to illustrate this better.